From d45fae589b8d8b6d36c211dcc46d767dda730b61 Mon Sep 17 00:00:00 2001
From: Andrew Cooper <andrew.cooper3@citrix.com>
Date: Mon, 12 Sep 2016 10:07:35 +0100
Subject: [PATCH] x86/cpu: Print CPU Family/Vendor infomation in both decimal
 and hexidecimal

Different manuals use different representations.

A new sample looks like:

(XEN) CPU Vendor: Intel, Family 6 (0x6), Model 60 (0x3c), Stepping 3 (raw 000306c3)

Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>
Acked-by: Jan Beulich <JBeulich@suse.com>
diff --git a/xen/arch/x86/cpu/common.c b/xen/arch/x86/cpu/common.c
index c00a0de..2127d8d 100644
--- a/xen/arch/x86/cpu/common.c
+++ b/xen/arch/x86/cpu/common.c
@@ -238,8 +238,9 @@ static void __init early_cpu_detect(void)
 	c->x86_capability[cpufeat_word(X86_FEATURE_SSE3)] = ecx;
 
 	printk(XENLOG_INFO
-	       "CPU Vendor: %s, Family %u, Model %u, Stepping %u (raw %08x)\n",
-	       this_cpu->c_vendor, c->x86, c->x86_model, c->x86_mask, eax);
+	       "CPU Vendor: %s, Family %u (%#x), Model %u (%#x), Stepping %u (raw %08x)\n",
+	       this_cpu->c_vendor, c->x86, c->x86,
+	       c->x86_model, c->x86_model, c->x86_mask, eax);
 
 	eax = cpuid_eax(0x80000000);
 	if ((eax >> 16) == 0x8000 && eax >= 0x80000008) {
